### ISSUE TITLE
When deactivate WooCommerce display error in Dashboard

### ISSUE THREAD

**Issue by niravcse006** [woocommerce/woocommerce#25123]
**Describe the bug**
In latest version first you can install and activate WooCommerce. After activate WooCommerce you can see the Activity dashboard widget in wordpress dashboard. After that when you deactive WooCommerce we can see some error in Activity dashboard widget.

**To Reproduce**
1. Activate WooCommerce
2. After that you can see Activity dashboard widget in wordpress dashboard.
3. After that you can go to plugin list.
4. Deactive WooCommerce
5. Go to WordPress Dashboard
6. See some error regarding Activity dashboard widget

**Screenshots**
1. https://ibb.co/MfvLrv3
2. https://ibb.co/gM419Bk

**Isolating the problem (mark completed items with an [x]):**
- I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- This bug happens with a default WooCommerce Storefront theme active(https://woocommerce.com/storefront/).
- I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
`
### WordPress Environment ###

WordPress address (URL): https://shallow-pigeon.jurassic.ninja
Site address (URL): https://shallow-pigeon.jurassic.ninja
WC Version: 3.8.0
REST API Version: ✔ 1.0.2
Log Directory Writable: ✔
WP Version: 5.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.41 (Unix) OpenSSL/1.0.2g
PHP Version: 7.3.12
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.28-0ubuntu0.16.04.2-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.8.0
WC Database Prefix: wp_b91cc6e39f_
MaxMind GeoIP Database: ✔
Total Database Size: 3.54MB
Database Data Size: 2.30MB
Database Index Size: 1.24MB
wp_b91cc6e39f_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_b91cc6e39f_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_b91cc6e39f_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_b91cc6e39f_options: Data: 1.48MB + Index: 0.06MB + Engine InnoDB
wp_b91cc6e39f_postmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_posts: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_b91cc6e39f_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_b91cc6e39f_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_b91cc6e39f_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_b91cc6e39f_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_b91cc6e39f_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_b91cc6e39f_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_b91cc6e39f_wc_order_stats: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_b91cc6e39f_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_b91cc6e39f_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_b91cc6e39f_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_b91cc6e39f_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (4) ###

Companion Plugin: by Osk – 1.11
WooCommerce Stripe Gateway: by WooCommerce – 4.3.1
WooCommerce Services: by Automattic – 1.22.1
WooCommerce: by Automattic – 3.8.0

### Inactive Plugins (5) ###

Akismet Anti-Spam: by Automattic – 4.1.3
Hello Dolly: by Matt Mullenweg – 1.7.2
Jetpack by WordPress.com: by Automattic – 7.9.1
WooCommerce - ShipStation Integration: by WooCommerce – 4.1.30
WooCommerce Admin: by WooCommerce – 0.22.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.5.3
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 3
Oldest: 2019-11-27 10:10:42 +0000
Newest: 2019-11-27 10:10:42 +0000

Pending: 1
Oldest: 2019-11-27 11:10:42 +0000
Newest: 2019-11-27 11:10:42 +0000

Canceled: 0
Oldest: –
Newest: –

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –

`
</details>

**Comment by juliaamosova** [comment]
Thank you for opening this issue @niravcse006. However, I can't reproduce it on my end. 

I followed the steps you provided but the Activity Dashboard widget didn't have any errors after deactivating WooCommerce. I've tried it on the brand new install - see the screencast below:

https://recordit.co/GFERVu10ip

Please let me know if I am missing a step. 

Have you tried the same on the brand new install as well? Or are there other plugins on your site?

**Comment by juliaamosova** [comment]
@niravcse006 another person reported the same issue (https://github.com/woocommerce/woocommerce/issues/25127) which I could reproduce. In order to replicate it, Setup Wizard needs to be completed and a product needs to be created. Since the other issue has more complete steps to reproduce it, I am closing yours in favor of https://github.com/woocommerce/woocommerce/issues/25127 - please continue to comment and discuss if needed there. Thanks!

**Comment by niravcse006** [comment]
Hello @juliaamosova 

**Have you tried the same on the brand new install as well? Or are there other plugins on your site?**
**Ans.** Please check above **Isolating the problem** and **WordPress Environment**

First of all, when I open this issue then you can't reproduce it on your end. After some time someone another person opens the same issue in some different way and after that, you can check again and find out this issue. 

Also, I don't understand why you can't able to find out this issue. Because here I check in your Jurassic ninja server and also I see that issue then why you don't see that?

**This is a site URL which is you have check-in:** https://defeated-raccoon.jurassic.ninja/

**Check this screencast:**
https://drive.google.com/file/d/1N9haW5C19y-_ZDeg2lrL2Oj3zZXJHobA/view

Also, my suggestion that, please don't close any issue without checking perfectly. And you aren't able to understand then please send a comment that I will not understand your issue so we will understand more deeply.

Thanks

**Comment by juliaamosova** [comment]
>Also, I don't understand why you can't able to find out this issue. Because here I check in your Jurassic ninja server and also I see that issue then why you don't see that?
>This is a site URL which is you have check-in: https://defeated-raccoon.jurassic.ninja/

@niravcse006 as you can see from the screencast that I posted (https://recordit.co/GFERVu10ip) in my first comment in this thread - I use https://defeated-raccoon.jurassic.ninja/ and follow the steps you provided. Clearly from that video, you can see that the issue is not there. 

It is there now because I used the same site (https://defeated-raccoon.jurassic.ninja/) to reproduce an issue that was reported later: https://github.com/woocommerce/woocommerce/issues/25127. Since that other issue had additional steps to reproduce the issue, I was able to see the error on that site. But only after I followed the additional steps. 

>Also, my suggestion that, please don't close any issue without checking perfectly. And you aren't able to understand then please send a comment that I will not understand your issue so we will understand more deeply.

We don't just close the issues when we can't reproduce it. You can search by `can't reproduce` label and see that there are [some issues open](https://github.com/woocommerce/woocommerce/issues?q=is%3Aissue+is%3Aopen+label%3A%22can%27t+reproduce%22). We always try to work with the author of the issue even if we can't reproduce it at first. 

I closed your issue because https://github.com/woocommerce/woocommerce/issues/25127 was opened shortly. Now we had 2 duplicate issues. In this case, we always close one of the issues to keep the repository clean. Usually, we would close the 2nd issue and keep the original. But in this case, the 2nd issue had steps to reproduce the issue where I could see the error, I closed yours. In any case, these 2 issues are duplicates. 

Once again, thank you for reporting.

**Comment by niravcse006** [comment]
<blockquote><p>I closed your issue because #25127 was opened shortly. Now we had 2 duplicate issues. In this case, we always close one of the issues to keep the repository clean. Usually, we would close the 2nd issue and keep the original. But in this case, the 2nd issue had steps to reproduce the issue where I could see the error, I closed yours. In any case, these 2 issues are duplicates.</p></blockquote>

@juliaamosova as you know that first, I opened this issue. Also, you know that if two same issues generate then the second duplicate issue closes because of to clean the repository. 

But in this case, you close the first generated issue because you can't able to find out the issue.

Also, I write in simple language with steps and also attach screenshots.

Thanks

**Comment by shamalisulakhe** [comment]
Hi @niravcse006 and @juliaamosova,

I don't know if I should enter in your conversation but just wanted to clarify the things.

Initially, I was not aware of your issue #25123. When I found the bug, what I found are the 'action_log' listed in the widget and also in the Dashboard->Comments and (not the notices since my wp_debug was off).

I checked existing WooCommerce issues first to see if there is any issue already created for it. I also came across this issue #25123, but since the issue title is 'WooCommerce display error in Dashboard', I thought that this issue refers to the notices ('_map_meta_cap was called incorrectly.._') displayed in the dashboard widget.

What my issue #25127 focuses on is the action_log getting treated as comments, and I found them listed in the Dashboard->Comments (wp-admin/edit-comments.php) as well. So I created a new issue for the same.

Thank you.
